src/LinuxTracing/LinuxTracingUtilsTest.cpp

```diff
@@ -56,14 +56,16 @@ TEST(GetThreadState, LinuxTracingTestsMainAndAnother) {
     // Make sure /proc/<pid>/stat is parsed correctly
     // even when the thread name contains spaces and parentheses.
     pthread_setname_np(pthread_self(), ") )  )()( )(  )");
+    // Give the main thread the time to get to the mutex.
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Have you considered locking the mutex `mutex` before actually creating the thread?

Reply:
Good point, I'll try and get back to you. One problem here is that the case that this one sleep should solve is by far the hardest to reproduce for me.

Reply:
Done.